include/nbl/video/IGPUCommandPool.h

```diff
@@ -31,11 +35,21 @@ class NBL_API IGPUCommandPool : public core::IReferenceCounted, public IBackendO
         // Vulkan: const VkCommandPool*
         virtual const void* getNativeHandle() const = 0;
 
+        virtual void reset()
+        {
+            assert(!getCreationFlags().hasFlags(ECF_RESET_COMMAND_BUFFER_BIT));
+            // Increase reset counter
+            m_resetCount.fetch_add(1);
+        }
+
     protected:
         virtual ~IGPUCommandPool() = default;
 
         core::bitflag<E_CREATE_FLAGS> m_flags;
         uint32_t m_familyIx;
+        std::atomic_uint32_t m_resetCount;
+
+        virtual uint32_t getResetCounter() { return m_resetCount.load(); }
```

Comment:
call `m_resetCount.load()` as late as possible when you need it. although this may be inlined and called when needed but let's be logical and return `std::atomic_uint32_t`
Also as I mentioned in the discord thread we could have even had normal uint instead of atomic. but since we decided to have atomics, we should return the counter as atomic and let user decide when to `.load` it